src/common/IterationLookup.ts

```diff
@@ -251,7 +251,9 @@ export class Iterations implements IterationLookup {
         ? DateTime.fromISO(dateString)
         : dateString;
     const monday = getMondayBefore(date);
-    const iterationString = this.byDate[monday.toISODate()];
+    const iterationString =
+      this.byDate[monday.toISODate()] ||
+      Object.values(this.byDate)[Object.values(this.byDate).length - 1];
```

Comment:
This is a generic method so it should return an undefined iteration if one doesn't exist. For example, this confounds the behavior of getAdjacentIteration. It's too cryptic for this to return valid iterations for invalid inputs. Handling this case should be done by the callers. Instead, you should just add a safety check.

```js
const iterationString = this.byDate[monday.toISODate()];
const iteration = this.byVersionString[iterationString];
return iteration ? {
  number: iterationString,
  start: iteration.startDate,
  due: iteration.endDate,
} : null;
```

and then change the jsdoc comment at the top so it says
```js
 * @returns {LegacyIteration|null}
```

and then add another method `getLatestIteration`

```js
/**
 * Get the latest iteration.
 * @returns {LegacyIteration}
 */
getLatestIteration(): LegacyIteration {
  const iterationString = this.orderedVersionStrings[
    this.orderedVersionStrings.length - 1
  ];
  const iteration = this.byVersionString[iterationString];
  return {
    number: iterationString,
    start: iteration && iteration.startDate,
    due: iteration && iteration.endDate,
  };
}
```

so the consumers can use `iterations.getIteration() ?? iterations.getLatestIteration()`

Only the UI consumers, though. Like, `getAdjacentIteration` should not be changed.

Reply:
I updated `getAdjacentIteration` to return `null` if `this.getIteration(dateString)` is null. Is that fine?

Reply:
Also, I use `getAdjacentIteration(-1, this.context.iterations.getLatestIteration().start)` for the current iteration rather than `getLatestIteration()` so I don't invalidate parts of Bugzy that refer to the "next iteration" in which case I use `getLatestIteration()`